src/main/java/com/watch/shop/app/controller/MenuHandler.java

```diff
@@ -0,0 +1,168 @@
+import com.watch.shop.app.view.WatchView;
+import lombok.RequiredArgsConstructor;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.util.EnumSet;
+import java.util.List;
+
+import static com.watch.shop.app.util.Constants.END_SEPARATOR;
+import static com.watch.shop.app.util.Constants.ENTER_ARRIVAL_DATE_MESSAGE;
+import static com.watch.shop.app.util.Constants.ENTER_BRAND_MESSAGE;
+import static com.watch.shop.app.util.Constants.ENTER_COLOR_MESSAGE;
+import static com.watch.shop.app.util.Constants.ENTER_MODEL_MESSAGE;
+import static com.watch.shop.app.util.Constants.ENTER_PRICE_MESSAGE;
+import static com.watch.shop.app.util.Constants.ENTER_TYPE_MESSAGE;
+import static com.watch.shop.app.util.Constants.FORMAT_HEADER;
+import static com.watch.shop.app.util.Constants.FORMAT_ROW;
+import static com.watch.shop.app.util.Constants.FORMAT_SEPARATOR;
+import static com.watch.shop.app.util.Constants.HAS_HEART_RATE_MONITOR_MESSAGE;
+import static com.watch.shop.app.util.Constants.IS_AUTOMATIC_MESSAGE;
+import static com.watch.shop.app.util.Constants.LUMINOUS_HANDS_MESSAGE;
+import static com.watch.shop.app.util.Constants.MAX_CHARGE_TIME_MESSAGE;
+import static com.watch.shop.app.util.Constants.MENU;
+import static com.watch.shop.app.util.Constants.OPERATION_SYSTEM_MESSAGE;
+import static com.watch.shop.app.util.Constants.POWER_RESERVE_MESSAGE;
+import static com.watch.shop.app.util.Constants.SEPARATOR;
+import static com.watch.shop.app.util.Constants.SORT_MENU;
+import static com.watch.shop.app.util.Constants.START_SEPARATOR;
+import static com.watch.shop.app.util.Constants.TOTAL_COST_MESSAGE;
+
+@RequiredArgsConstructor
+public class MenuHandler {
+
+    public static final String BRAND = "Brand";
+    public static final String MODEL = "Model";
+    public static final String COLOR = "Color";
+    public static final String TYPE = "Type";
+    public static final String PRICE = "Price";
+    public static final String ARRIVAL_DATE = "Arrival Date";
+
+    private final WatchView watchView;
+    private final InputHandler inputHandler = new InputHandler();
+
+    public String printMainMenu() {
+        watchView.printMessage(MENU);
+
+        return inputHandler.readLineFromConsole();
+    }
+
+    public String printSortMenu() {
+        watchView.printMessage(SORT_MENU);
+
+        return inputHandler.readLineFromConsole();
+    }
+
+    public void displayAllWatches(List<Watch> watches) {
+        watchView.printMessage(START_SEPARATOR);
+        watchView.printFormattedMessage(FORMAT_HEADER, BRAND, MODEL, COLOR, TYPE, PRICE, ARRIVAL_DATE);
+        watchView.printFormattedMessage(FORMAT_SEPARATOR, SEPARATOR.repeat(13), SEPARATOR.repeat(27), SEPARATOR.repeat(12), SEPARATOR.repeat(12), SEPARATOR.repeat(12), SEPARATOR.repeat(14));
+
+        watches.forEach(w -> watchView.printFormattedMessage(FORMAT_ROW, w.getBrand(), w.getModel(), w.getColor(), w.getType(), w.getPrice(), w.getArrivalDate()));
```

Comment:
when a lot of arguments (4 and more) it's more convenient to refactor when it vertical:
w -> watchView.printFormattedMessage(FORMAT_ROW, w.getBrand(),
                                                                                 w.getModel(), 
                                                                                 w.getColor(), 
                                                                                w.getType(), 
                                                                                w.getPrice(), 
                                                                                w.getArrivalDate()));
                                                                                
or use private helper method:
watches.forEach(this::printFormattedWatch);

...
private void printFormattedWatch(Watch watch) {
   view.printFormattedMessage(FORMAT_ROW, watch.getBrand(),
          watch.getModel() ...
          
          }
          
          2. Also w -> w.getBrand
           w is not good shortage for variable. In Java allowed common known shortages, k, v for key, value map, or i -> integer etc.
           
           w it's not obviously clear what is it.
           
           watches.forEach(watch ->  view.printFormattedMessage....
           
           Of cources, in more technical code, in lambda allowed usage of first letter, but when logic is less complex and more algorythmic.
           
           In this case watch - > is ok